server/sonar-server/src/test/java/org/sonar/elasticsearch/test/EsTestCluster.java

```diff
@@ -0,0 +1,965 @@
+    DeleteIndexResponse response = builder.get();
+    MatcherAssert.assertThat("Delete Index failed - not acked", response.isAcknowledged(), CoreMatchers.equalTo(true));
+  }
+
+  /**
+   * Removes all templates, except the templates defined in the exclude
+   */
+  public void wipeAllTemplates(Set<String> exclude) {
+    if (size() > 0) {
+      GetIndexTemplatesResponse response = client().admin().indices().prepareGetTemplates().get();
+      for (IndexTemplateMetaData indexTemplate : response.getIndexTemplates()) {
+        if (exclude.contains(indexTemplate.getName())) {
+          continue;
+        }
+        try {
+          client().admin().indices().prepareDeleteTemplate(indexTemplate.getName()).execute().actionGet();
+        } catch (IndexTemplateMissingException e) {
+          // ignore
+        }
+      }
+    }
+  }
+
+  /**
+   * Deletes index templates, support wildcard notation.
+   * If no template name is passed to this method all templates are removed.
+   */
+  public void wipeTemplates(String... templates) {
+    if (size() > 0) {
+      // if nothing is provided, delete all
+      if (templates.length == 0) {
+        templates = new String[] {"*"};
+      }
+      for (String template : templates) {
+        try {
+          client().admin().indices().prepareDeleteTemplate(template).execute().actionGet();
+        } catch (IndexTemplateMissingException e) {
+          // ignore
+        }
+      }
+    }
+  }
+
+  /**
+   * Deletes repositories, supports wildcard notation.
+   */
+  public void wipeRepositories(String... repositories) {
+    if (size() > 0) {
+      // if nothing is provided, delete all
+      if (repositories.length == 0) {
+        repositories = new String[] {"*"};
+      }
+      for (String repository : repositories) {
+        try {
+          client().admin().cluster().prepareDeleteRepository(repository).execute().actionGet();
+        } catch (RepositoryMissingException ex) {
+          // ignore
+        }
+      }
+    }
+  }
+
+  /**
+   * Runs the code block for 10 seconds waiting for no assertion to trip.
+   */
+  public static void assertBusy(Runnable codeBlock) throws Exception {
+    assertBusy(codeBlock, 10, TimeUnit.SECONDS);
+  }
+
+  /**
+   * Runs the code block for the provided interval, waiting for no assertions to trip.
+   */
+  private static void assertBusy(Runnable codeBlock, long maxWaitTime, TimeUnit unit) throws Exception {
+    long maxTimeInMillis = TimeUnit.MILLISECONDS.convert(maxWaitTime, unit);
+    long iterations = Math.max(Math.round(Math.log10(maxTimeInMillis) / Math.log10(2)), 1);
+    long timeInMillis = 1;
+    long sum = 0;
+    List<AssertionError> failures = new ArrayList<>();
+    for (int i = 0; i < iterations; i++) {
+      try {
+        codeBlock.run();
+        return;
+      } catch (AssertionError e) {
+        failures.add(e);
+      }
+      sum += timeInMillis;
+      Thread.sleep(timeInMillis);
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Remove this use of "Thread.sleep()". [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS2925)